common/src/main/java/org/wso2/testgrid/common/util/TestGridUtil.java

```diff
@@ -152,13 +155,32 @@ public static String getTestGridHomePath() {
         return testGridHomePath.toAbsolutePath().toString();
     }
 
+    /**
+     * Returns the directory location where the test run artifacts resides.
+     *
+     * @param testPlan test plan for getting the test run artifacts location
+     * @return path of the test run artifacts
+     */
+    public static Path getTestRunArtifactsDirectory(TestPlan testPlan) {
+        DeploymentPattern deploymentPattern = testPlan.getDeploymentPattern();
+        Product product = deploymentPattern.getProduct();
+        int testRunNumber = testPlan.getTestRunNumber();
+
+        String productDir = StringUtil.concatStrings(product.getName(), "_", product.getVersion()
+                , "_", product.getChannel());
+        String deploymentDir = deploymentPattern.getName();
+        String infraDir = getInfraParamUUID(testPlan.getInfraParameters());
+
+        return Paths.get(productDir, deploymentDir, infraDir, String.valueOf(testRunNumber));
+    }
+
     /**
      * Returns a UUID specific to the infra parameters.
      *
      * @param infraParams infra parameters to get the UUID
      * @return UUID specific to the infra parameters
      */
-    public static String getInfraParamUUID(String infraParams) {
+    private static String getInfraParamUUID(String infraParams) {
         return UUID.nameUUIDFromBytes(infraParams.getBytes(Charset.defaultCharset())).toString();
```

Comment:
In here shouldn't we check for same infra parameters in different forms,
like `ubuntu/mysql/jdk8` and `UBUNTU/MySQL/JDK8` since we have no control over these parameter values.